### ISSUE TITLE
🚀 Support for `additionalFiles` at the target level

### ISSUE THREAD

**Issue by danyf90** [tuist/tuist#3574]
In our feature targets we have some additional files we would like to add to the Xcode project (e.g. READMEs, CHANGELOGs, etc.).

Right now we can only add them in the `ProjectDescription.Project`, but this causes problems with focusing, as the files will be included also if the target is not among the focused ones.

It should be quite straightforward to add the additionalFiles property also to `ProjectDescription.Target`, do you see any problem with that?

**Comment by pepibumur** [comment]
:shipit: :shipit!

**Comment by kwridan** [comment]
Would be handy to list out some use cases where `additionalFiles` are associated with a target rather than a project.

The `additionalFiles` feature was first discussed / introduced in https://github.com/tuist/tuist/issues/202. The rationale for having it defined at the project and workspace level is that those files don't belong to any target per se _(at least at the Xcode level, they are merely file references in the project)_. 

Some examples of where they are used for today:

- `README`
- `Documentation` folder reference
- Custom `.modulemap` files
- Bridging headers files which are only referenced in build settings, and not included in the `Headers` build phase

Going through just those I can see some value in also including target level `additionalFiles`  as the last two examples are ones that logically have an associated with a target. I wouldn't deprecated the Project attribute as the first two use cases logically belong to a project as opposed to an individual target.

**Comment by danyf90** [comment]
I agree there are cases in which makes sense to add them to the project (e.g. "global" README or documentation), and cases in which makes sense to add them to the target (e.g. target specific README, modulemap, bridging headers, etc.)
